src/distribution_spatial.cpp

```diff
@@ -180,6 +182,70 @@ Position SphericalIndependent::sample(uint64_t* seed) const
   return {x, y, z};
 }
 
+//==============================================================================
+// MeshSpatial implementation
+//==============================================================================
+
+MeshSpatial::MeshSpatial(pugi::xml_node node)
+{
+  // No in-tet distributions implemented, could include distributions for the barycentric coords
+  // Read in unstructured mesh from mesh_id value
+  int32_t mesh_id = std::stoi(get_node_value(node, "mesh_id"));
+  // Get pointer to spatial distribution
+  mesh_idx_ = model::mesh_map.at(mesh_id);
+  
+  // Check whether mesh pointer points to a mesh
+  mesh_ptr_ = dynamic_cast<Mesh*>(model::meshes[mesh_idx_].get());
+  if (!mesh_ptr_) {fatal_error("Mesh passed to spatial distribution is not a mesh object"); }
+
+  // Initialize arrays for CDF creation
+  tot_bins_ = mesh_ptr_->n_bins();
+  std::vector<double> strengths = {};
+  strengths.resize(tot_bins_);
+  double temp_total_strength = 0.0;
+  mesh_CDF_.resize(tot_bins_+1);
+  mesh_CDF_[0] = {0.0};
+  total_strength_ = 0.0;
+  mesh_strengths_.resize(tot_bins_);
+
+  // Create cdfs for sampling for an element over a mesh
+  // Volume scheme is weighted by the volume of each tet
+  // File scheme is weighted by an array given in the xml file
+
+  mesh_strengths_ = std::vector<double>(tot_bins_, 1.0);
+  if (check_for_node(node, "strengths")) {
+      strengths = get_node_array<double>(node, "strengths");
+      if (strengths.size() != mesh_strengths_.size()){
+        fatal_error("Number of entries in the strength matrix does not match the number of mesh entities");
+      }
+      mesh_strengths_ = strengths;
+  }
+
+  if (get_node_value_bool(node, "volume_normalized")) {
+    for (int i = 0; i < tot_bins_; i++) {
+      mesh_strengths_[i] *= mesh_ptr_->volume(i);
+    }
+  }
+
+  for (int i = 0; i<tot_bins_; i++){
+    temp_total_strength = temp_total_strength + mesh_strengths_[i];
+  }
+  total_strength_ = temp_total_strength;
+  for (int i = 0; i<tot_bins_; i++){
+    mesh_CDF_[i+1] = mesh_CDF_[i] + mesh_strengths_[i]/total_strength_;
+  }
+}
+
+Position MeshSpatial::sample(uint64_t* seed) const
+{ 
+  // Create random variable for sampling element from mesh
+  float eta = prn(seed);
+  // Sample over the CDF defined in initialization above
+  int32_t tet_bin = lower_bound_index(mesh_CDF_.begin(), mesh_CDF_.end(), eta);
+  return mesh_ptr_->sample(seed, tet_bin);
```

Comment:
```suggestion
  int32_t elem_idx = lower_bound_index(mesh_CDF_.begin(), mesh_CDF_.end(), eta);
  return mesh_ptr_->sample(seed, elem_idx);
```
Just making names more general for other non-tet `MeshSpatial` types